### PR TITLE
Detox, part 1 of N

### PR DESCRIPTION
**Pull request by DMRobertson**
As part of #11537 we are going to move away from tox.

This pulls out two linting steps from tox which refer to the `scripts` directory. Will be useful for #12118.

### CODE REVIEW

**Comment by DMRobertson** [comment]
To be convinced this change works, I will
- edit the sample config and news file so they fail the checks. Commit, let CI fail.
- revert the last commit. Let CI pass (hopefully).

**Review by clokep — APPROVED** [review]
Seems fine overall.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0
       - uses: actions/setup-python@v2
-      - run: pip install tox
+      - run: "pip install 'towncrier>=18.6.0rc1'"
```
[the remark this answers is not in the document]
Comment (DMRobertson) [reply .github/workflows/tests.yml:53]: I'd like to stick the version guard in setup.py too, but I'd prefer to leave that for when I get further with transitioning to poetry.

[the remark this answers is not in the document]
Comment (DMRobertson) [reply .github/workflows/tests.yml:53]: And as for just installing here: I was trying to reproduce tox's behaviour as closely as I could (for now).

[the remark this answers is not in the document]
Comment (clokep) [reply .github/workflows/tests.yml:53]: Fair enough!

**Pull request merged**